### PR TITLE
Addcopy

### PR DESCRIPTION
**Pull request by vapaavetehinen**
reworked copy for visual indication of copy

### CODE REVIEW

File: src/App.svelte [hunk 1]
```diff
@@ -1,7 +1,12 @@
 <script>
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
-  console.log(exampleData);
+  export let grabIP = function(text){
+    const ips = document.getElementsByClassName('ip');
+    const toCopy = Array.from(ips).filter((ip) => {return ip.value === text})[0];
+    toCopy.select();
+    document.execCommand("copy");
+  }
```
Comment (akhilcacharya) [inline src/App.svelte]: Why did you remove navigator usage? Why not have a fallback?

File: src/App.svelte [hunk 2]
```diff
@@ -1,7 +1,12 @@
 <script>
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
-  console.log(exampleData);
+  export let grabIP = function(text){
+    const ips = document.getElementsByClassName('ip');
+    const toCopy = Array.from(ips).filter((ip) => {return ip.value === text})[0];
+    toCopy.select();
```
Comment (akhilcacharya) [inline src/App.svelte]: You should use find() instead of filter. 

also destructure the IP and just return the internal value ```find(({ value }) => value ==ip)``` or something

**Review by akhilcacharya — CHANGES_REQUESTED** [review]

File: src/App.svelte (hunk 1 above)
Comment (vapaavetehinen) [reply src/App.svelte]: I don't think if this fails the nav is gonna work.

File: src/App.svelte (hunk 2 above)
Comment (vapaavetehinen) [reply src/App.svelte]: find is a good idea, but the whole reason for not just returning the text is to highlight the text in the box. That's the whole point of this rewriting.

File: src/ExampleData.js [hunk 3]
```diff
@@ -2,191 +2,191 @@ export const exampleData = [
   {
     banner: "//cdn.minecraft-server-list.com/serverlogo/268353.jpg",
     name: "LemonCloud Minecraft Server",
-    ip: "play.lemoncloud.net"
+    domain: "play.lemoncloud.net"
```
Comment (peakay) [inline src/ExampleData.js]: this should probably be address, all references to domain should also be address

**Review by peakay — CHANGES_REQUESTED** [review]

File: src/App.svelte (hunk 1 above)
Comment (vapaavetehinen) [reply src/App.svelte]: happy to be wrong though, feel free to link some documentation that suggests otherwise.

File: src/App.svelte (hunk 2 above)
Comment (vapaavetehinen) [reply src/App.svelte]: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard

File: src/App.svelte (hunk 1 above)
Comment (vapaavetehinen) [reply src/App.svelte]: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard

File: src/App.svelte (hunk 2 above)
Comment (akhilcacharya) [reply src/App.svelte]: Find would return the node itself, unless i'm misunderstanding you

Comment (vapaavetehinen) [reply src/App.svelte]: yeah I changed it from filter to find, maybe I'm not understanding what you mean by destructing it beforehand.

File: src/App.svelte [hunk 4]
```diff
@@ -2,8 +2,8 @@
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
   export let grabDomain = function(text){
-    const domains = document.getElementsByClassName('domain');
-    const toCopy = Array.from(domains).filter((domain) => {return domain.value === text})[0];
```
Comment (akhilcacharya) [inline src/App.svelte]: .filter(({ value }) => value === text)

File: src/App.svelte [hunk 5]
```diff
@@ -1,7 +1,12 @@
 <script>
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
-  console.log(exampleData);
+  export let grabDomain = function(text){
+    const addresses = document.getElementsByClassName('address');
+    const toCopy = Array.from(addresses).find(({value}) => value === text);
```
Comment (ming08108) [inline src/App.svelte]: Instead of grabbing all of the address fields you should be binding the element. The copy element could even be its own component.

https://svelte.dev/tutorial/bind-this

File: src/ExampleData.js [hunk 6]
```diff
@@ -2,191 +2,191 @@ export const exampleData = [
   {
     banner: "//cdn.minecraft-server-list.com/serverlogo/268353.jpg",
     name: "LemonCloud Minecraft Server",
-    ip: "play.lemoncloud.net"
+    address: "play.lemoncloud.net"
```
Comment (ming08108) [inline src/ExampleData.js]: change address back to ip

**Review by ming08108 — CHANGES_REQUESTED** [review]

File: src/CopyInput.svelte [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+let inputfield;
+export let ip;
+
+let grabIp = function(){
+    inputfield.select()
+    document.execCommand("copy");
+  }
+</script>
+
+<input bind:this={inputfield} value={ip} readonly><button  on:click={grabIp}>Copy</button>
```
Comment (ming08108) [inline src/CopyInput.svelte]: you have an extra space here

**Comment by akhilcacharya** [comment]
It seems to work fine, but could look better. Lets tackle in a follow up.

**Comment by vapaavetehinen** [comment]
> It seems to work fine, but could look better. Lets tackle in a follow up.

Yes. This was already decided.

**Comment by vapaavetehinen** [comment]
whoops

**Review by ming08108 — APPROVED** [review]
